marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -775,6 +775,35 @@ declare function hent:json-schema-generate($entity-title as xs:string, $uber-mod
   return xdmp:to-json($uber-model)
 };
 
+declare %private function hent:add-indexes-for-structured-entity-properties($entity-path as xs:string, $definiton-name as xs:string, $definitions as map:map, $primary-entity as map:map) {
+  let $entity-type := map:get($definitions, $definiton-name)
+  let $entity-type-properties := map:get($entity-type, "properties")
+
+  let $_ :=
+    for $entity-type-property in map:keys($entity-type-properties)
+    let $ref := map:get($entity-type-properties, $entity-type-property)=>map:get("$ref")
+    let $items := map:get($entity-type-properties, $entity-type-property)=>map:get("items")
+    return
+      if (fn:empty($ref) and (fn:empty($items) or fn:not(fn:starts-with(map:get($items, "$ref"), "#")))) then
+        if (map:get($entity-type-properties, $entity-type-property)=>map:get("facetable")) then
+          json:array-push(map:get($primary-entity, "rangeIndex"), $entity-path || "/" || $entity-type-property)
+        else ()
+      else
+        let $definiton-name :=
+          if (fn:empty($items)) then
+            fn:substring-after($ref, "#/definitions/")
+          else
+            fn:substring-after(map:get($items, "$ref"), "#/definitions/")
+        let $_ :=
+          if (fn:not(fn:not($definiton-name))) then
+            let $path := $entity-path || "/" || $entity-type-property || "/" || $definiton-name
+            let $_ := hent:add-indexes-for-structured-entity-properties($path, $definiton-name, $definitions, $primary-entity)
+            return ()
+          else ()
+        return ()
+  return $primary-entity
```

Comment:
The function name starts with "add", which typically implies that there's no return value - i.e. you're adding something to something else that already exists.